### PR TITLE
Continuous ml 2

### PR DESCRIPTION
**Pull request by HenrydeB**
trying this again fresh

### CODE REVIEW

File: .github/workflows/cml.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - uses: iterative/setup-cml@v1
+
+      - name: Train model
+        run: |
+          pip install poetry
+          poetry install
```
Comment (msoria17) [inline .github/workflows/cml.yaml]: this can be poetry install --no-dev

File: mlops_team_project/models/xgboost_model.py [hunk 2]
```diff
@@ -183,7 +173,14 @@ def model(
 
     logging.info(classification_report(y_test, preds, target_names=target_names))
 
-    save_model_to_google(model)
+    if os.environ.get('RUN_CML') == 'true':
+        creat_cml_report(y_test, preds, target_names)
+
+    with open("models/xgboost_model.pkl", "wb") as file:
```
Comment (msoria17) [inline mlops_team_project/models/xgboost_model.py]: why do we need to save the model locally?

File: pyproject.toml [hunk 3]
```diff
@@ -17,9 +17,13 @@ rich = "^13.3.2"
 wandb = "^0.17.0"
 torch = "^2.3.0"
 snakeviz = "^2.2.0"
+fastapi = "^0.111.0"
```
Comment (msoria17) [inline pyproject.toml]: we don't need fasty api or uvicorn in here

**Comment by HenrydeB** [comment]
classification_report.txt

**Comment by HenrydeB** [comment]
precision    recall  f1-score   support

non-diabetic       0.78      0.70      0.74      7198
diabetic       0.72      0.79      0.75      6941

    accuracy                           0.75     14139

macro avg       0.75      0.75      0.75     14139
weighted avg       0.75      0.75      0.75     14139

**Review by msoria17 — APPROVED** [review]
lgtm

**Comment by HenrydeB** [comment]
precision    recall  f1-score   support

non-diabetic       0.78      0.70      0.74      7198
diabetic       0.72      0.79      0.75      6941

    accuracy                           0.75     14139

macro avg       0.75      0.75      0.75     14139
weighted avg       0.75      0.75      0.75     14139

**Comment by HenrydeB** [comment]
precision    recall  f1-score   support

non-diabetic       0.78      0.70      0.74      7198
diabetic       0.72      0.79      0.75      6941

    accuracy                           0.75     14139

macro avg       0.75      0.75      0.75     14139
weighted avg       0.75      0.75      0.75     14139

**Pull request merged**